src/web/utils/errorHandler.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { ErrorInfo, ReactNode } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import { ApiError } from './apiError';
+
+import './errorHandler.scss';
+
+interface ErrorState {
+  errorId: string;
+  errorHash?: string;
+  errorCount: number;
+  hasError?: boolean;
+  message?: string;
+  stack?: string;
+  componentStack?: string;
+}
+
+export interface RenderedErrorProps {
+  errorId?: string;
+  errorHash?: string;
+  errorCount: number;
+  message?: string;
+  stack?: string;
+  componentStack?: string;
+}
+
+export function analyticsIdentifier(errorId?: string, errorHash?: string) {
+  if (!errorHash && !errorId) return 'No error identifier';
+  return errorHash
+    ? `error hash: ${errorHash}` // From backend
+    : `error ID: ${errorId}`; // Generated by frontend to make it possible to triage user reported errors.
+}
+
+export function ErrorView({
+  message,
+  errorId,
+  errorHash,
+}: {
+  message: string;
+  errorId?: string;
+  errorHash?: string;
+}) {
+  return (
+    <div className='error-content'>
+      <img alt='Error icon' src='uid2-logo.png' />
+      <div>Error</div>
+      <div>{message}</div>
+      <div>({analyticsIdentifier(errorId, errorHash)})</div>
+    </div>
+  );
+}
+export interface ErrorBoundaryPropType {
+  children: ReactNode;
+  [key: string]: unknown;
+}
+
+/** Create an Error Boundary around a given component.
+ * @param ErrorComponent - Custom error component used to display the error.
+ */
+export function errorHandler<P extends ErrorBoundaryPropType>(
+  ErrorComponent: React.ComponentType<RenderedErrorProps>
+) {
+  class ErrorHandlerWrapperComponent extends React.Component<P, ErrorState> {
```

Comment:
Is this borrowed from somewhere else? If so, can you document where it came from? I'd prefer to avoid class-based components but if it's saving some time, maybe it's reasonable.

Reply:
Yeah, I don't like class components as well, but sadly react error boundaries currently mandate it [here](https://legacy.reactjs.org/docs/error-boundaries.html#introducing-error-boundaries)
```
Only class components can be error boundaries. In practice, most of the time you’ll want to declare an error boundary component once and use it throughout your application.
```



(I implemented this recently in creatives, which in-turn is borrowed from ad-platform. Do you think its okay to reference ttd-codebase from here?)

Reply:
Maybe don't mention the TTD codebase directly - I was mostly curious :)

Reply:
Open source contributors can't see that code so it's not really useful to have a reference to it.